### PR TITLE
164797151 User comment Feature

### PR DESCRIPTION
**Pull request by oldmonad**
#### What does this PR do?
* Have user comment route working

#### Description of Task to be completed?
* This PR makes sure a user can make a comment on an article and also reply another comment under a previous comment in a single thread

#### How should this be manually tested?
* clone `https://github.com/andela/zinnia-ah-backend.git`
* checkout to  a new branch = `ft-comments-endpoint-164797151`
* pull from the remote branch to update the local branch
* run `npm install`
* run `sudo npm install -g pg`
* run `cp .env.example .env`
* create a database
* fill in the details in your `.env`
* run `sequelize db:migrate`
* run npm test
* open Postman and test route after inputing required parameters by entering POST `http://localhost:3000/api/v1/articles/articleId/comments` for creating a comment and `http://localhost:3000/api/v1/articles/articleId/comments/commentId/thread`

#### What are the relevant pivotal tracker stories?(if applicable)
* #164797151

#### Screenshots

### CODE REVIEW

File: src/routes/articles.route.js [hunk 1]
```diff
@@ -49,6 +54,69 @@ const articleRouter = Router();
  *       500:
  *         description: ran
  */
-articleRouter.post('/', create);
+articleRouter.post('/', createArticle);
+import checkAuthorizedUser from './middlewares/authorizedUser';
```
Comment (Japhethca) [inline src/routes/articles.route.js]: All our import statements should be at the top, let move it there.

File: src/routes/controllers/comments.controller.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      return errorResponse(res, 404, 'User does not exist');
+    }
+
+    const newComment = await Comment.create({
+      userId: id,
+      articleId,
+      body: comment,
+    });
+
+    if (newComment) {
```
Comment (Japhethca) [inline src/routes/controllers/comments.controller.js]: Am not sure there is any need for this `if` check here, you can just return newComment in the response.

File: src/routes/controllers/comments.controller.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { errorResponse, successResponse } from '../../utils/helpers.utils';
+
+import models from '../../db/models';
+
+const { Comment, User, Article } = models;
+
+export const createComments = async (req, res) => {
```
Comment (Japhethca) [inline src/routes/controllers/comments.controller.js]: this should be `createComment`

File: src/routes/controllers/comments.controller.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      return errorResponse(res, 404, 'This comment does not exist');
+    }
+
+    const newThreadedComment = await Comment.create({
+      userId: id,
+      articleId,
+      body: threadedComment,
+      parentId: commentId,
+    });
+    if (newThreadedComment) {
```
Comment (Japhethca) [inline src/routes/controllers/comments.controller.js]: we can remove this `if` condition.

**Review by Japhethca — CHANGES_REQUESTED** [review]
Just a few comments.

File: src/routes/controllers/comments.controller.js (hunk 3 above)
Comment (oldmonad) [reply src/routes/controllers/comments.controller.js]: I have implemeted this change

File: src/routes/controllers/comments.controller.js (hunk 2 above)
Comment (oldmonad) [reply src/routes/controllers/comments.controller.js]: I have removed this if comment

**Review by Japhethca — APPROVED** [review]

**Pull request merged**